### PR TITLE
integrate fresh

### PR DESCRIPTION
**Pull request by Uzlopak**
#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: lib/SendStream.js [hunk 1]
```diff
@@ -226,6 +224,74 @@ SendStream.prototype.isConditionalGET = function isConditionalGET () {
...
+    return false
+  }
+
+  // if-none-match
+  if ('if-none-match' in req.headers) {
+    const ifNoneMatch = req.headers['if-none-match']
+
+    if (ifNoneMatch === '*') {
+      return true
+    } else {
```
Comment (kibertoad) [inline lib/SendStream.js:247]: we return anyway, else is redundant

File: lib/SendStream.js [hunk 2]
```diff
@@ -226,6 +224,74 @@ SendStream.prototype.isConditionalGET = function isConditionalGET () {
...
+      const etag = res.getHeader('etag')
+
+      if (typeof etag !== 'string') {
+        return false
+      }
+
+      const etagL = etag.length
+      const isMatching = parseTokenList(ifNoneMatch, function (match) {
+        const mL = match.length
```
Comment (kibertoad) [inline lib/SendStream.js:256]: can we do external pure function instead? inline ones cause deoptimization, AFAIR

File: lib/parseTokenList.js [hunk 3]
```diff
@@ -7,10 +7,12 @@
  * @private
  */
 
-function parseTokenList (str) {
+const slice = String.prototype.slice
```
Comment (kibertoad) [inline lib/parseTokenList.js:10]: we destructure slice and not invoke instance method?

File: lib/SendStream.js (hunk 1 above)
Comment (Uzlopak) [reply lib/SendStream.js:247]: If I remove this return true, the test "where "If-None-Match" is set" "should respond with 306 Not Modified when using wildcard * on existing file" will fail.

File: lib/parseTokenList.js (hunk 3 above)
Comment (Uzlopak) [reply lib/parseTokenList.js:10]: slice is ambigous for node. Is it slice of string or slice of array. so it is potential deoptimization issue. So to make it easier for node, you can define them in variable and call them directly.

File: lib/SendStream.js (hunk 1 above)
Comment (kibertoad) [reply lib/SendStream.js:247]: return is fine, else is unnecessary

File: lib/SendStream.js (hunk 2 above)
Comment (Uzlopak) [reply lib/SendStream.js:256]: How should i inject etag variable into that external pure function?

File: lib/SendStream.js (hunk 1 above)
Comment (Uzlopak) [reply lib/SendStream.js:247]: ah right

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by Uzlopak** [comment]
I investigated further and did not see any perf regressions.

**Pull request merged**